lib/mauth/authenticators/authenticator.rb

```diff
@@ -0,0 +1,135 @@
+# methods common to RemoteRequestAuthenticator and LocalAuthenticator
+
+module MAuth
+  class Client
+    module Authenticator
+      ALLOWED_DRIFT_SECONDS = 300
+
+      # takes an incoming request or response object, and returns whether
+      # the object is authentic according to its signature.
+      def authentic?(object)
+        log_authentication_request(object)
+        begin
+          authenticate!(object)
+          true
+        rescue InauthenticError, MauthNotPresent
+          false
+        end
+      end
+
+      # raises InauthenticError unless the given object is authentic. Will only
+      # authenticate with v2 if the environment variable AUTHENTICATE_WITH_ONLY_V2
+      # is set. Otherwise will authenticate with only the highest protocol version present
+      def authenticate!(object)
+        if authenticate_with_only_v2? && authentication_present_v1(object)
+          msg = 'This service requires mAuth v2 mcc-authentication header but only v1 x-mws-authentication is present'
+          raise MAuth::MissingV2Error, msg
```

Comment:
This is not rescued above while the other types are, is that on puporse?

Reply:
It was so that I could rescue `MAuth::MissingV2Error` [here](https://github.com/mdsol/mauth-client-ruby/pull/25/files#diff-f77b06d19ecda5de902d19ad6367fb8dR28) in the rack middleware and respond appropriately. I wanted to avoid replicating the logic in the middleware of when to raise a missing v2 error but I guess `authenticate?` should really only respond true or false.

Reply:
Updated to rescue `MAuth::MissingV2Error`s and added logic to rack middleware for when we should return the missing v2 response.